reframe/frontend/loader.py

```diff
@@ -114,6 +118,17 @@ def prefix(self):
     def recurse(self):
         return self._recurse
 
+    def _set_defaults(self, test_registry):
+        for test in test_registry:
+            for name, val in self._external_vars.items():
+                if '.' in name:
+                    testname, varname = name.split('.', maxsplit=1)
```

Comment:
The `.` in the name, as we set it here, would not set an attribute in the nested object. It won't act as an operator. It will just create an attribute at the class named `f.a`. Of course this attribute will only be accessible through `__dict__`. In any case, the change that you propose below will cover that as well, so I will not explicitly look for a single `.`.

Reply:
Yeah, all this is covered by the code below, so all good here from my side.